DotnetFoundation/DotnetFoundation.Api/Controllers/TaskDetailsController.cs

```diff
@@ -0,0 +1,147 @@
+
+        response.Data = await _TaskDetailsService.GetActiveTasksAsync().ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Get task details by Id.
+    /// </summary>
+    /// <param name="taskId">Id of task record</param>
+    [HttpGet("{taskId}")]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<TaskDetailsResponse>>> GetTaskByIdAsync(int taskId)
+    {
+        BaseResponse<TaskDetailsResponse> response = new(ResponseStatus.Fail);
+
+        response.Data = await _TaskDetailsService.GetTaskByIdAsync(taskId).ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Add new task.
+    /// </summary>
+    /// <param name="detailRequest">Role request details</param>
+    [HttpPost]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<string>>> InsertTaskAsync(TaskDetailsRequest detailRequest)
+    {
+        BaseResponse<string> response = new(ResponseStatus.Fail);
+
+        response.Data = await _TaskDetailsService.InsertTaskAsync(detailRequest).ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Update details of a task when the Id is passed.
+    /// </summary>
+    /// <param name="taskId">Id of task record</param>
+    /// <param name="modifiedDetails">Modified details for task record</param>
+    [HttpPut("{taskId}")]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<string>>> UpdateTaskAsync(int taskId, TaskDetailsRequest modifiedDetails)
+    {
+        BaseResponse<string> response = new(ResponseStatus.Fail);
+
+        response.Data = await _TaskDetailsService.UpdateTaskAsync(taskId, modifiedDetails).ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Change status of task to inactive.
+    /// </summary>
+    /// <param name="taskId">Id of task record</param>
+    [HttpPut("inactive/{taskId}")]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<string>>> InactiveTaskAsync(int taskId)
```

Comment:
This can be a DELETE HTTP action.
It doesn't need any action name like "inactive"
The return type should be `BaseResponse<DTO>`. It should return the DTO object that was deactivated.

Reply:
Made it as DELETE and removed inactive keyword